docs: rewrite all documentation

domWalters opened the pull request.

domWalters commented:
@hkato, sneak peek on the documentation rewrite I've been working on today:

https://mkdocs-to-pdf.readthedocs.io/en/latest/

Critique appreciated.

hkato commented:
@domWalters 

I submitted a PR #46 to the readme branch. but please excuse my English.

hkato commented:
I think it's fine to search the README. And the GitHub appearance is not so good.

```yaml
---
search:
  exclude: true
---
```

hkato commented:
Could you please add my name to the CHANGELOG? similar to other contributors.

domWalters commented:
> @domWalters
> 
> I submitted a PR #46 to the readme branch. but please excuse my English.

Merged.

domWalters commented:
> I'm wondering why the `docs_dir` is in the root. (The README.md links to the `docs` directory).
> 
> ```diff
> --- a/mkdocs.yml
> +++ b/mkdocs.yml
> @@ -1,5 +1,4 @@
>  ---
> -docs_dir: .
>  repo_name: domWalters/mkdocs-to-pdf
>  repo_url: https://github.com/domWalters/mkdocs-to-pdf
>  site_author: Dominic Walters
> @@ -7,9 +6,6 @@ site_dir: site/
>  site_name: mkdocs-to-pdf
>  site_url: https://mkdocs-to-pdf.readthedocs.io
>  
> -exclude_docs: |
> -  samples/
> -
>  markdown_extensions:
>    - pymdownx.highlight:
>        anchor_linenums: true
> @@ -23,11 +19,11 @@ markdown_extensions:
>  
>  nav:
>    - Home: README.md
> -  - Installation: docs/installation.md
> -  - Usage: docs/usage.md
> -  - Changelog: docs/CHANGELOG.md
> -  - License: docs/LICENSE.md
> -  - Stuff to Migrate: docs/stuff-to-migrate.md
> +  - Installation: installation.md
> +  - Usage: usage.md
> +  - Changelog: CHANGELOG.md
> +  - License: LICENSE.md
> +  - Stuff to Migrate: stuff-to-migrate.md
>  
>  plugins:
>    - minify:
> ```

You're right, this is needlessly complicated.

My thinking was that I want the homepage of the site to be the exact same as the README. To accomplish that, the README needs to be in the docs directory.

I've changed to just symlink it into the docs like CHANGELOG and LICENSE

domWalters commented:
> I think it's fine to search the README. And the GitHub appearance is not so good.
> 
> ```yaml
> ---
> search:
>   exclude: true
> ---
> ```

Agreed, didn't catch that.

domWalters commented:
@hkato, can you do another review pass: https://mkdocs-to-pdf.readthedocs.io/en/latest/

In particular, can you check that the CJK characters are still okay on the site. I can see them in the PDF but my browser gives me squares due to my font not supporting the characters.

Main updates;

- Made changes that you suggested.
- Moved `samples/material` to `docs/examples`, so all of the tests you wrote now go onto the site.
- Site now provides its PDF.
  - This required me to change how `requirements-dev.txt` was generated.
  - This also allows any prospective user to immediately get a PDF with all of the examples in it.
- Did a pass over the dependencies and removed some unnecessary ones.
  - This this was because `mkdocs-material` now depends on them itself.

hkato commented:
@domWalters 

>In particular, can you check that the CJK characters are still okay on the site. I can see them in the PDF but my browser gives me squares due to my font not supporting the characters.

LGTM

squares = [ToFu](https://en.wikipedia.org/wiki/Tofu) => **No To**Fu fonts => Noto fonts (designed by Google/Adobe)

hkato commented:
I pushed 2 commits, and it successfully tested my permissions.

- mkdocs-pdf-export-plugin -> `mkdocs-pdf-export-plugin`
- Requirements section
    - In the future, we need to add Chromium/Chrome based browsers

hkato reviewed (approved):
LGTM 👍
